Update NuGet package and Sample to .NET Core 3.1

normj opened the pull request:
![image](https://user-images.githubusercontent.com/1653751/124050489-82273f80-d9cf-11eb-875e-9ba50c34f1f4.png)


*Description of changes:*
The Amazon.AspNetCore.Identity.Cognito was targeting .NET Standard 2.0 and .NET Core 3.0. Since .NET Core 3.0 is out of support I updated the .NET Core 3.0 to 3.1. A major version bump was set to signify the potential breaking change but I don't expect much .NET Core 3.0 usage.

The sample in the repo had the following updates:
* Changed target to .NET Core 3.1
* Fixed Forget Password scenario
* Removed unnecessary UI components that were left over from the original ASP.NET Core project template
* Reordered appsettings.json file to have user pool id before client since that is the order you create the AWS resources
* Updated the index page to have links to useful next step documentations


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

boblodgett commented on samples/Samples/Pages/Index.cshtml:
```diff
@@ -4,93 +4,30 @@
+<h1>Amazon Cognito ASP.NET Core Web Application Sample</h1>
+
+This sample application demonstrates how to integrate Amazon Cognito as an identity provider for an ASP.NET Core Web Application. The sample adds the <a href="https://www.nuget.org/packages/Amazon.AspNetCore.Identity.Cognito/" target="_blank">Amazon.AspNetCore.Identity.Cognito</a>
+NuGet package as the registered identity service provider for the ASP.NET Core application. The sample also add sample login pages that demonstrate how customize ASP.NET Core login process to use Amazon Cognito.
```
"The sample also add sample login pages that demonstrate how customize ASP.NET Core login process to use Amazon Cognito."

=>

"The sample also adds sample login pages that demonstrate how to customize the ASP.NET Core login process to use Amazon Cognito."

normj replied:
Updated

boblodgett commented on src/Amazon.AspNetCore.Identity.Cognito/Amazon.AspNetCore.Identity.Cognito.csproj:
```diff
@@ -16,22 +16,22 @@
-    <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp3.0'">
-      <DefineConstants>NETCOREAPP_3_0</DefineConstants>
+    <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+      <DefineConstants>NETCOREAPP_3_1</DefineConstants>
```
"NETCOREAPP3_1" should already be defined so that we do not need to define our own: https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-a-target-framework

boblodgett commented on src/Amazon.AspNetCore.Identity.Cognito/CognitoKeyNormalizer.cs:
```diff
@@ -37,7 +37,7 @@ public string Normalize(string key)
 #endif
 
-#if NETCOREAPP_3_0
+#if NETCOREAPP_3_1
```
As mentioned in the csproj we should be using the built in preprocessor symbols here and in the other files: NETCOREAPP3_1

normj replied:
Fixed

ppittle reviewed (commented):
I followed the steps in  **Setup.md** and added a `AWS` to the **appsettings.json** file.  When I hit play in VS to launch IISExpress, I get an `InternalServerException`:

```
UriFormatException: Invalid URI: The hostname could not be parsed.
System.Uri.CreateThis(string uri, bool dontEscape, UriKind uriKind)
System.Uri..ctor(string uriString)
Amazon.Runtime.ClientConfig.GetUrl(RegionEndpoint regionEndpoint, string regionEndpointServiceName, bool useHttp, bool useDualStack)
Amazon.Runtime.ClientConfig.DetermineServiceURL()
Amazon.Runtime.Internal.DefaultRetryPolicy..ctor(IClientConfig config)
Amazon.Runtime.AmazonServiceClient.BuildRuntimePipeline()
Amazon.Runtime.AmazonServiceClient..ctor(AWSCredentials credentials, ClientConfig config)
Amazon.CognitoIdentityProvider.AmazonCognitoIdentityProviderClient..ctor(AWSCredentials credentials, AmazonCognitoIdentityProviderConfig clientConfig)
```

**Is there more setup required?**

normj commented:
> I followed the steps in **Setup.md** and added a `AWS` to the **appsettings.json** file. When I hit play in VS to launch IISExpress, I get an `InternalServerException`:
> 
> ```
> UriFormatException: Invalid URI: The hostname could not be parsed.
> System.Uri.CreateThis(string uri, bool dontEscape, UriKind uriKind)
> System.Uri..ctor(string uriString)
> Amazon.Runtime.ClientConfig.GetUrl(RegionEndpoint regionEndpoint, string regionEndpointServiceName, bool useHttp, bool useDualStack)
> Amazon.Runtime.ClientConfig.DetermineServiceURL()
> Amazon.Runtime.Internal.DefaultRetryPolicy..ctor(IClientConfig config)
> Amazon.Runtime.AmazonServiceClient.BuildRuntimePipeline()
> Amazon.Runtime.AmazonServiceClient..ctor(AWSCredentials credentials, ClientConfig config)
> Amazon.CognitoIdentityProvider.AmazonCognitoIdentityProviderClient..ctor(AWSCredentials credentials, AmazonCognitoIdentityProviderConfig clientConfig)
> ```
> 
> **Is there more setup required?**
> 
> Found the problem [1-SETUP.md](https://github.com/aws/aws-aspnet-cognito-identity-provider/blob/192a82bd3833b8b84ea1892c5c275dea16fcad1a/docs/1-Setup.md) incorrectly says:
> 
> > Just make the necessary changes to the following properties to the _appsettings.json_ file
> 
> [README.md](https://github.com/aws/aws-aspnet-cognito-identity-provider/blob/normj/update-sample/README.md) has the correct instructions:
> 
> > For a development user pool edit either the `appsettings.Development.json` file or the projects [secrets.json](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-5.0&tabs=windows) file.
> 
> After updating the _stub_ in `appsettings.Development.json` I was able to launch the website correctly.
> 
> **Can you updated [1-SETUP.md](https://github.com/aws/aws-aspnet-cognito-identity-provider/blob/192a82bd3833b8b84ea1892c5c275dea16fcad1a/docs/1-Setup.md) to match the verbiage in [README.md](https://github.com/aws/aws-aspnet-cognito-identity-provider/blob/normj/update-sample/README.md)**.

Updated 1-Setup.md file to use appsettings.Development.json

iofluxdev1 commented:
Legend, so glad I found this PR! Resolves issues that has been around for a year with password resetting. 

`var code = await _userManager.GeneratePasswordResetTokenAsync(user);`

used to work then one day stopped working and now it seems 

`await user.ForgotPasswordAsync();`

is the way to go.
